### PR TITLE
Add Taskcluster support to run Lint tests automatically

### PR DESCRIPTION
**Pull request by Standard8**
This hooks up taskcluster support so that we can run the Lint tests automatically.

There's a docker file as well, with instructions on how to regenerate it if necessary.

r? @felipc

### CODE REVIEW

**Comment by Standard8** [comment]
FWIW I'm currently working on this, but eslint-plugin-mozilla has been giving me some pain, hopefully I'll get this sorted sometime next week.

**Comment by Standard8** [comment]
@felipc I've now updated this to get it passing again.

The addition of the testing/ and tools/ files are to mirror mozilla-central. Hopefully I'll find a better way in the future, but for now, this gets us going. Given the lint issues that have shown up, I think it is better to get the automation running here asap.

File: addons/aushelper/bootstrap.js [hunk 1]
```diff
@@ -55,7 +57,7 @@ function startup() {
     wrk = Cc["@mozilla.org/windows-registry-key;1"].createInstance(Ci.nsIWindowsRegKey);
     wrk.open(wrk.ROOT_KEY_LOCAL_MACHINE, regCPUPath, wrk.ACCESS_READ);
   } catch (e) {
-    Cu.reportError("AUSHelper - unable to open registry. Exception: " + e);
+    Cu.reportError(`AUSHelper - unable to open registry. Exception: ${e}`);
```
Comment (felipc) [inline addons/aushelper/bootstrap.js]: Why these changes? Is this a requirement due to an eslint rule? If yes I don't think this particular rule should be enabled :)

File: addons/aushelper/bootstrap.js [hunk 2]
```diff
@@ -35,16 +35,18 @@ const observer = {
...
         TelemetryLog.log("WEBSENSE_PREF_CHANGED", [prefValue]);
         break;
+      }
     }
   }
 };
 
-function startup() {
+/* eslint max-depth:off */
+function startup() { // eslint-disable-line complexity, max-statements
```
Comment (felipc) [inline addons/aushelper/bootstrap.js]: do you know why max-depth, complexity, max-statements are being triggered here?

Comment (Standard8) [inline addons/aushelper/bootstrap.js]: The startup() function is too complex and needs breaking down - since I'm not actually working on the add-on, I didn't really want to do that work.

File: addons/aushelper/bootstrap.js (hunk 1 above)
Comment (Standard8) [inline addons/aushelper/bootstrap.js]: This is the [prefer-template rule](http://eslint.org/docs/rules/prefer-template) - it is new in es6, I didn't like it at first, but now I'm getting used to it, I find it a bit clearer than string concatenation.

**Comment by Standard8** [comment]
@felipc Thanks. I've also added a simple commit in to fix issues in a newly added file so that the tests pass.

I've decided to leave prefer-template rule in for now, and we can see how it goes. At some stage I want to start unifying the rules with what we have for mozilla-central, so I'll review it then.
